### ISSUE TITLE
Change default root url

### ISSUE THREAD

**Issue by sepiropht** [zeit/next.js#2289]
I have an next/js app behind nginx with custom server  node/express . 
I launch it to port 3000. Nginx redirect the request myserver.com/mynextapp/ to port 3000 , but this is not working. 
I can see the page server rendered, but the js doesn't working . In browser tool network  i got 404 response  for app.js and page/, because  the request is myserver.com/_next/228ef92e59d376f055fc2c6d01c93b82/app.js instead of myserver.com/mynextapp/_next/228ef92e59d376f055fc2c6d01c93b82/app.js
It worked well when i simply opened the port 3000 in myserver.com but i try to avoid that.
Any idea ?
Thanks

**Comment by timneutkens** [comment]
This is a duplicate of https://github.com/zeit/next.js/issues/257. Please see that issue for some community solutions 👍

**Comment by qwang06** [comment]
I ran into this issue as well and none of the solutions in #257 seem to resolve this.

I ended up modifying the server/index.js file and updating the routes in the method defineRoutes:
`
      Object.keys(routes).map(route => {
        routes[this.renderOpts.assetPrefix + route] = routes[route];
      });
`

I know this isn't the best solution right now but it's the only thing that works. Hopefully support for something like this is available in the future so I don't have to straight up modify core code.
